crates/common/tedge_config_macros/impl/src/query.rs

```diff
@@ -401,29 +513,31 @@ fn generate_string_writers(paths: &[VecDeque<&FieldOrGroup>]) -> TokenStream {
         })
         .multiunzip();
 
+        // TODO do we need an _ branch on these?
     quote! {
         impl TEdgeConfigDto {
-            pub fn try_update_str(&mut self, key: WritableKey, value: &str) -> Result<(), WriteError> {
+            pub fn try_update_str(&mut self, key: &WritableKey, value: &str) -> Result<(), WriteError> {
                 match key {
                     #(#update_arms)*
                 };
                 Ok(())
             }
 
-            pub fn unset_key(&mut self, key: WritableKey) {
+            pub fn try_unset_key(&mut self, key: &WritableKey) -> Result<(), WriteError> {
```

Comment:
This can now fail as `WritableKey` is no longer guaranteed to point to a key that definitely exists (when named `c8y`s are enabled).